### PR TITLE
Styles-Objects

### PR DESCRIPTION
**Pull request by EvgenBabin1**

### CODE REVIEW

File: src/convertToObject.js [hunk 1]
```diff
@@ -3,16 +3,37 @@
...
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const string = sourceString.split(';');
```
Comment (Yurii837) [inline src/convertToObject.js:18]: Maybe rename because it is array.

**Review by Yurii837 — APPROVED** [review]
Good!

File: src/convertToObject.js [hunk 2]
```diff
@@ -3,16 +3,37 @@
...
+  const values = [];
+  const keys = [];
+  const styles = {};
+
+  string.forEach(item => {
+    const property = item.split(':');
+
+    values.push(property[0].trim());
+    keys.push(property[1]);
+  });
```
Comment (AlinaMatuschak) [inline src/convertToObject.js:28]: can you here add a property to styles?

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]

File: src/convertToObject.js [hunk 3]
```diff
@@ -3,16 +3,36 @@
...
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const array = {};
```
Comment (AlinaMatuschak) [inline src/convertToObject.js:18]: I thought that it's object )

File: src/convertToObject.js [hunk 4]
```diff
@@ -3,16 +3,36 @@
...
+  let values = [];
+
+  const expressions = sourceString.split(';')
+    .map(str => str.trim())
+    .filter(trimmed => trimmed.length > 0);
+
+  expressions.forEach(exp => {
+    parts = exp.split(':');
+    key = parts[0].trim();
+    values = parts[1].trim();
```
Comment (AlinaMatuschak) [inline src/convertToObject.js:30]: value`s`? but here just one string

File: src/convertToObject.js [hunk 5]
```diff
@@ -3,16 +3,36 @@
...
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const array = {};
+  let parts = '';
+  let key = [];
+  let values = [];
```
Comment (AlinaMatuschak) [inline src/convertToObject.js:21]: Do you need here these variables?

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**